getuser fallback fails on Python 3.13+

### Summary

I'm using ansible-test to run some tests on my private collection. Those tests are run in a container, where there's no user entry in `/etc/passwd` matching the current `uid`. Among those tests, some are using the `PluginLoader`.

At some point, the `PluginLoader` runs this code  (`utils/display.py:178`, `class FilterUserInjector`):

```python
try:
    username = getpass.getuser()
except KeyError:
    # people like to make containers w/o actual valid passwd/shadow and use host uid
    username = 'uid=%s' % os.getuid()
```
As you can see, there's a mechanism to deal with such uids.


But, if you look at `getpass.py` from the stdlib:

it was (py 3.10):
```python
def getuser():
    """[...]"""
    for name in ('LOGNAME', 'USER', 'LNAME', 'USERNAME'):
        user = os.environ.get(name)
        if user:
            return user

    # If this fails, the exception will "explain" why
    import pwd
    return pwd.getpwuid(os.getuid())[0]
```

and now with py 3.13:
```python
def getuser():
    """[...]"""
    for name in ('LOGNAME', 'USER', 'LNAME', 'USERNAME'):
        user = os.environ.get(name)
        if user:
            return user

    try:
        import pwd
        return pwd.getpwuid(os.getuid())[0]
    except (ImportError, KeyError) as e:
        raise OSError('No username set in the environment') from e
```

The `KeyError` is now an `OSError`, which is not catched, so the tests fail.

Solutions:
-  catch OSError as well as KeyError in `utils/display.py`
-  or, add `LOGNAME` and/or `USER` and/or `LNAME` and/or `USERNAME` to the list of optional env vars in `ansible-test` when resetting the environment. `HOME` is already part of those required/optional vars, why not those ? 

### Issue Type

Bug Report

### Component Name

lib/utils/display.py

### Ansible Version

```console
$ ansible --version
ansible [core 2.19.3]
  config file = None
  configured module search path = ['/tmp/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /venv/lib/python3.13/site-packages/ansible
  ansible collection location = /tmp/.ansible/collections:/usr/share/ansible/collections
  executable location = /venv/bin/ansible
  python version = 3.13.9 (main, Oct 21 2025, 02:15:13) [GCC 14.2.0] (/venv/bin/python)
  jinja version = 3.1.6
  pyyaml version = 6.0.3 (with libyaml v0.2.5)
```

### Configuration

```console
# if using a version older than ansible-core 2.12 you should omit the '-t all'
$ ansible-config dump --only-changed -t all
CONFIG_FILE() = None

GALAXY_SERVERS:
```

### OS / Environment

docker container: lib/python-3.13.9-slim
host: darwin 24.6.0 arm64

Same behaviour can be observed on any python 3.13 containerized environment.

### Steps to Reproduce

`docker run -ti -u 10042 -e USER=foobar -e HOME=/tmp -v $PWD/app -w /app python:3.13.9-slim ansible-test units --local --python 3.13`

have a test file in $PWD/app whith something like this:

```python
from ansible.plugins.loader import lookup_loader

class TestLookupModule(TestCase):
    def setUp(self):
        self.lookup = lookup_loader.get("foo.bar")

    def test_foo(self):
        self.lookup.run([], [], {})
```

### Expected Results

I expect the tests to behave OK.

### Actual Results

```console
Tests fail with an `OSError` exception:


==================================== ERRORS ====================================
____________ ERROR collecting tests/unit/plugins/lookup/test_foo.py _____________
/usr/local/lib/python3.13/getpass.py:173: in getuser
    return pwd.getpwuid(os.getuid())[0]
           ^^^^^^^^^^^^^^^^^^^^^^^^^
E   KeyError: 'getpwuid(): uid not found: 10042'

The above exception was the direct cause of the following exception:
tests/unit/plugins/lookup/test_foo.py:16: in <module>
    from ansible.plugins.loader import lookup_loader
/tmp/ansible-test-31pvuidz/ansible/plugins/__init__.py:29: in <module>
    from ansible.utils.display import Display
/tmp/ansible-test-31pvuidz/ansible/utils/display.py:172: in <module>
    class FilterUserInjector(logging.Filter):
/tmp/ansible-test-31pvuidz/ansible/utils/display.py:180: in FilterUserInjector
    username = getpass.getuser()
               ^^^^^^^^^^^^^^^^^
/usr/local/lib/python3.13/getpass.py:175: in getuser
    raise OSError('No username set in the environment') from e
E   OSError: No username set in the environment
------------------------------- Captured stdout --------------------------------
```

### Code of Conduct

- [x] I agree to follow the Ansible Code of Conduct

--- a/changelogs/fragments/getuser-exception-handling.yml
+++ b/changelogs/fragments/getuser-exception-handling.yml
@@ -1,0 +1,4 @@
+bugfixes:
+  - display - Fix ``getuser`` fallback error handling on Python 3.13 and later.
+    (https://github.com/ansible/ansible/issues/86142)
+  - local connection - Fix ``getuser`` fallback error handling on Python 3.13 and later.

--- a/lib/ansible/plugins/connection/local.py
+++ b/lib/ansible/plugins/connection/local.py
@@ -67,7 +67,8 @@ class Connection(ConnectionBase):
         self.cwd = None
         try:
             self.default_user = getpass.getuser()
-        except KeyError:
+        except (ImportError, KeyError, OSError):
+            # deprecated: description='only OSError is required for Python 3.13+' python_version='3.12'
             display.vv("Current user (uid=%s) does not seem to exist on this system, leaving user empty." % os.getuid())
             self.default_user = ""
 

--- a/lib/ansible/utils/display.py
+++ b/lib/ansible/utils/display.py
@@ -174,7 +174,8 @@ class FilterUserInjector(logging.Filter):
 
     try:
         username = getpass.getuser()
-    except KeyError:
+    except (ImportError, KeyError, OSError):
+        # deprecated: description='only OSError is required for Python 3.13+' python_version='3.12'
         # people like to make containers w/o actual valid passwd/shadow and use host uids
         username = 'uid=%s' % os.getuid()
 

